src/Build/Graph/ProjectInterpretation.cs

```diff
@@ -323,7 +324,7 @@ public GlobalPropertyPartsForMSBuildTask AddPropertyToUndefine(string propertyTo
 
             var globalPropertyParts = globalPropertyModifiers?.Aggregate(defaultParts, (currentProperties, modifier) => modifier(currentProperties, projectReference)) ?? defaultParts;
 
-            if (globalPropertyParts.AllEmpty())
+            if (globalPropertyParts.AllEmpty() && !DynamicPlatformEnabled)
```

Comment:
Should this be || ?

Reply:
No, when we use dynamic platform resolution we never want to resuse a global props dictionary. we want to create a copy. otherwise if project A(x86) references project B(AnyCPU) since we reuse the global property dictionary project A will have global properties (Platform:AnyCPU)